### ISSUE TITLE
app.command doesn't register command when running on AWS Lambda

### ISSUE THREAD

**Issue by azizmb** [slackapi/bolt-python#369]
When trying to register command using bolt-python, the command doesn't trigger when deployed on lambda. Testing locally via the websockets this works as expected, but, as mentioned, fails on lambda with API gateway.

The request url does end with `/slack/events`. 

### Reproducible in:

```bash
pip freeze | grep slack
python --version
sw_vers && uname -v # or `ver`
```

#### The `slack_bolt` version
slack-bolt==1.6.1
slack-sdk==3.6.0

#### Python runtime version
Python 3.8.8

#### OS info

#59-Ubuntu SMP Fri Apr 16 08:43:28 UTC 2021

#### Steps to reproduce:

API Gateway configured via CDK:
```python
bolt_lambda = lmbda_python.PythonFunction(
    self,
    'bolt-lambda',
    runtime=lmbda.Runtime.PYTHON_3_8,
    entry='my-bot',
    environment={
        'SLACK_BOT_TOKEN': "...",
        'SLACK_SIGNING_SECRET': "...",
    },
)

api = apigateway.LambdaRestApi(
    self,
    'bolt-api-gateway',
    handler=bolt_lambda
)
```

```python
# Handler
app = App(process_before_response=True, raise_error_for_unhandled_request=True)

@app.command('/my-command')
def my_command(ack, say, command, logger):
    # Acknowledge command request
    ack()
    say(f"say {command['text']}")


SlackRequestHandler.clear_all_log_handlers()
logging.basicConfig(format="%(asctime)s %(message)s", level=logging.DEBUG)


def handler(event, context):
    slack_handler = SlackRequestHandler(app=app)
    return slack_handler.handle(event, context)
```

### Expected result:
Running `/my-command hello world` should respond with `hello world`

### Actual result:
Slackbot responds with:
`/my-command failed with the error "dispatch_failed"`

I see that the request is reaching the lambda from the logs:

```2021-06-07 18:26:57,387 Incoming event: {'resource': '/{proxy+}', 'path': '/slack/events', 'httpMethod': 'POST', 'headers': {'Accept': 'application/json,*/*', 'Accept-Encoding': 'gzip,deflate', 'CloudFront-Forwarded-Proto': 'https', 'CloudFront-Is-Desktop-Viewer': 'true', 'CloudFront-Is-Mobile-Viewer': 'false', 'CloudFront-Is-SmartTV-Viewer': 'false', 'CloudFront-Is-Tablet-Viewer': 'false', 'CloudFront-Viewer-Country': 'US', 'Content-Type': 'application/x-www-form-urlencoded', 'Host': '....execute-api.eu-west-1.amazonaws.com', 'User-Agent': 'Slackbot 1.0 (+https://api.slack.com/robots)', 'Via': '1.1 e4c06b6e6eb895470e2fd65bbc93b3b6.cloudfront.net (CloudFront)', 'X-Amz-Cf-Id': '...', 'X-Amzn-Trace-Id': '...', 'X-Forwarded-For': '...', 'X-Forwarded-Port': '443', 'X-Forwarded-Proto': 'https', 'X-Slack-Request-Timestamp': '1623090416', 'X-Slack-Signature': '...'}, 'multiValueHeaders': {'Accept': ['application/json,*/*'], 'Accept-Encoding': ['gzip,deflate'], 'CloudFront-Forwarded-Proto': ['https'], 'CloudFront-Is-Desktop-Viewer': ['true'], 'CloudFront-Is-Mobile-Viewer': ['false'], 'CloudFront-Is-SmartTV-Viewer': ['false'], 'CloudFront-Is-Tablet-Viewer': ['false'], 'CloudFront-Viewer-Country': ['US'], 'Content-Type': ['application/x-www-form-urlencoded'], 'Host': ['....execute-api.eu-west-1.amazonaws.com'], 'User-Agent': ['Slackbot 1.0 (+https://api.slack.com/robots)'], 'Via': ['1.1 e4c06b6e6eb895470e2fd65bbc93b3b6.cloudfront.net (CloudFront)'], 'X-Amz-Cf-Id': ['...'], 'X-Amzn-Trace-Id': ['...'], 'X-Forwarded-For': ['...'], 'X-Forwarded-Port': ['443'], 'X-Forwarded-Proto': ['https'], 'X-Slack-Request-Timestamp': ['1623090416'], 'X-Slack-Signature': ['...']}, 'queryStringParameters': None, 'multiValueQueryStringParameters': None, 'pathParameters': {'proxy': 'slack/events'}, 'stageVariables': None, 'requestContext': {'resourceId': '...', 'resourcePath': '/{proxy+}', 'httpMethod': 'POST', 'extendedRequestId': '...', 'requestTime': '07/Jun/2021:18:26:56 +0000', 'path': '/prod/slack/events', 'accountId': '822242914934', 'protocol': 'HTTP/1.1', 'stage': 'prod', 'domainPrefix': 'y4rmpptk64', 'requestTimeEpoch': 1623090416577, 'requestId': 'e33ce95b-4a73-4774-8050-c9becca1f70a', 'identity': {'cognitoIdentityPoolId': None, 'accountId': None, 'cognitoIdentityId': None, 'caller': None, 'sourceIp': '...', 'principalOrgId': None, 'accessKey': None, 'cognitoAuthenticationType': None, 'cognitoAuthenticationProvider': None, 'userArn': None, 'userAgent': 'Slackbot 1.0 (+https://api.slack.com/robots)', 'user': None}, 'domainName': '....execute-api.eu-west-1.amazonaws.com', 'apiId': '...'}, 'body': 'token=...&team_id=...&team_domain=...&channel_id=...&channel_name=directmessage&user_id=...&user_name=...&command=%2Fmy-command&text=hello-world&api_app_id=...&is_enterprise_install=false&response_url=...', 'isBase64Encoded': False}, context: <__main__.LambdaContext object at 0x7f6b8c921d00>```

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

**Comment by azizmb** [comment]
After spending a day on this, I finally figured it out. The error was on my end, the environment variables I had set belonged to a different app.

While not a bug, but, nothing in the documentation pointed me to the fact that the tokens were incorrect. An enhancement perhaps?

**Comment by mwbrooks** [comment]
Hi @azizmb,

Thanks a bunch for the detailed issue and steps to reproduce! Helping with AWS Gateway/Lambda can be tricky, so your API Gateway configuration is VERY appreciated!

I'm sorry to hear that you burned an entire day on this issue 🙁  I talked with a few of the maintainers on whether we can enhance Bolt or the documentation to help guide others.

At the moment, Bolt will log inactive tokens, but there is no way for us to detect that Bolt is using another app's token (or to do a parity check between the tokens and signing secrets). So on the client-side of Bolt, we don't have a lot of flexibility.

What I can do is:

1. Take a look over the Bolt documentation and add a hint for developers to double-check their tokens. When juggling multiple environments (e.g. "My App (dev)" and "My App (staging)") it can be easy to mix-up tokens. If I send a pull request, I'll reference this GitHub issue!
2. I'll file an internal issue with the Slack API server-side team and point to this GitHub issue. I can't guarantee that it'll land on their roadmap, but it helps build a use-case grounded in real-world usage 💪🏻

**Comment by azizmb** [comment]
@mwbrooks thanks a lot for the response.

I understand that it might be tricky to detect that the tokens belong to a different app, but maybe some sort of authentication failure error? Because the logs were ok, so I assumed it had to be something code related. I stumbled onto the fact that the tokens were wrong because I had verified everything else multiple times and I thought let me just confirm.

On a side note, thanks for the great work, love slack and using bots for automation is getting addictive 😀.
